lib/src/presentation/widget/drawer.dart

```diff
@@ -0,0 +1,57 @@
+import 'package:flutter/material.dart';
+
+import '../../core/util/constants.dart';
+import 'drawer_list_tile.dart';
+
+class MyDrawer extends StatefulWidget {
+  const MyDrawer({
+    super.key,
+  });
+
+  @override
+  State<MyDrawer> createState() => _MyDrawerState();
+}
+
+class _MyDrawerState extends State<MyDrawer> {
+  final List<String> routesNameInDrawer = [
+    'Home',
+    'Popular Movies',
+    'Upcoming Movies',
+  ];
+  final Image logoImage = Image.asset(
+    fit: BoxFit.fill,
+    ConstantsClass.logoImagePath,
+  );
+  final double drawerOpacity = 0.9;
+  @override
+  Widget build(
+    BuildContext context,
+  ) {
+    return Opacity(
+      opacity: drawerOpacity,
+      child: Drawer(
+        backgroundColor: ConstantsClass.appBarColor,
+        child: ListView(
+          children: [
+            logoImage,
+            DrawerListTile(
+              icon: Icons.home,
+              text: routesNameInDrawer[0],
+              routeToGo: ConstantsClass.homeRouteName,
+            ),
```

Comment:
You can wrapped the icon, route title and route in a enum class, like this:
```
enum Route {
      home(Icons.home, 'Home', Constants.homeRoute,),
      popular(Icons.stars_rounded, 'Popular Movies', Constants.popularMoviesRoute,),
      upcoming(Icons.arrow_forward_outlined, 'Upcoming Movies', Constants.upcomingMoviesRoute);
      
      const Routes(this.icon, this.routeName, this.route,);
      
      final Icon icon;
      final String routeName;
      final String route;
}
```

Reply:
Also review the file name, ConstanClass could be renamed to Constants

Reply:
solved